### PR TITLE
Options refactoring and docs update

### PR DESCRIPTION
**Pull request by sseraj**
## Purpose
This PR refactors the code to:
1) Automatically generate the options documentation
2) Make the pyHyp class inherit from BaseSolver in baseclasses

The option type and value checking in BaseSolver breaks backward compatibility for pyHyp runscripts, most commonly in two ways:
1) String-type values are now case sensitive. This mainly comes up for `fileType` and `outerFaceBC`.
2) The distinction between `float` and `int` types is now enforced. This is commonly an issue for ``marchDist``, `ps0`, `pGridRatio`, and `cmax`.

See the changes to the example scripts in this PR for examples. The actual mesh generation is unchanged.

There are a few runscripts elsewhere in our code that will have to be updated. I will open PRs for cgnsUtilities and our tutorials shortly. Once this is merged, I will make a new release and make these incompatibilities clear in the release notes.

In addition to the options table, I updated some outdated information and scripts in the docs. I also renamed the tutorials so that they more accurately represent the content.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [x] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Testing
See the [BaseSolver tests](https://github.com/mdolab/baseclasses/blob/master/tests/test_BaseSolver.py) for examples on how the type and value checking works.

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

File: pyhyp/pyHyp.py [hunk 1]
```diff
@@ -555,7 +410,7 @@ def __init__(self, comm=None, options=None, debug=False):
         # Convert file type to integer
         fileType = {"cgns": self.hyp.hypinput.cgnsfiletype, "plot3d": self.hyp.hypinput.plot3dfiletype}
 
-        intFileType = fileType[self._go("fileType").lower()]
+        intFileType = fileType[self._go("fileType")]
```
Comment (nwu63) [inline pyhyp/pyHyp.py:413]: Only thing from me is whether we want to throw away the `self._go` function and instead just say `self.getOption()`, which is far more explicit and the extra characters are probably not a big deal.

Comment (sseraj) [reply pyhyp/pyHyp.py:413]: I agree that using `self.getOption()` is better. I'll change that.

**Comment by nwu63** [comment]
Oh also Travis was complaining that `examples/BWB/runBWB.py` need to be formatted.

**Review by nwu63 — APPROVED** [review]

File: pyhyp/pyHyp.py [hunk 2]
```diff
@@ -685,20 +540,90 @@ def __init__(self, comm=None, options=None, debug=False):
             intFileType = self.hyp.hypinput.patchinput
 
         # Now run the fortran setup.
-        self.hyp.setup(self._go("inputFile"), intFileType)
+        self.hyp.setup(self.getOption("inputFile"), intFileType)
+
+    @staticmethod
```
Comment (joanibal) [inline pyhyp/pyHyp.py:545]: Nice use of a decorator

File: pyhyp/pyHyp.py [hunk 3]
```diff
@@ -544,9 +400,8 @@ def __init__(self, comm=None, options=None, debug=False):
         if options is None:
             raise Error("The options = keyword argument is *NOT* optional. " "It must always be provided")
 
-        # Setup the options
-        self.options = {}
-        self._checkOptions(options, defaultOptions)
+        # Initialize the inherited BaseSolver
+        super().__init__(name, category, defOpts, options)
```
Comment (joanibal) [inline pyhyp/pyHyp.py:404]: This is so much cleaner :+1:

**Review by joanibal — APPROVED** [review]

**Pull request merged**